### PR TITLE
Remove legacy `Result` parameters

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Removes: `early_stop_on`, `checkpoint_on`, `hiddens` from `Result`

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Comment by Borda** [comment]
@carmocca how is it going here? :]

**Comment by carmocca** [comment]
I have some BC concerns about this and this is not high priority. Moving it to next week

File: pytorch_lightning/trainer/supporters.py [hunk 1]
```diff
@@ -104,21 +104,6 @@ def _agg_memory(self, how: str):
                 return getattr(self.memory[:self.current_idx], how)()
 
 
-class Accumulator(object):
```
Comment (ananthsub) [inline pytorch_lightning/trainer/supporters.py:107]: what was the history behind this? why do we have this and the tensor accumulator above?

File: pytorch_lightning/trainer/training_loop.py [hunk 2]
```diff
@@ -373,11 +366,10 @@ def _process_training_step_output_1_0(self, training_step_output, split_batch):
         # handle scalar return
         elif isinstance(training_step_output, torch.Tensor):
             loss = training_step_output
-            result["extra"] = {}
 
         # map to results under the hood
         result.minimize = loss
-        result.hiddens = hiddens
+        self.trainer.hiddens = hiddens
```
Comment (ananthsub) [inline pytorch_lightning/trainer/training_loop.py:372]: is there any other spot we can stick this state to attaching more properties to the trainer?

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/trainer/supporters.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/trainer/supporters.py:107]: No idea about this one, it predates my presence in the project.

`TensorAccumulator` is used for the progress bar running (loss)

https://github.com/PyTorchLightning/pytorch-lightning/blob/0e45220263f4e2045dfe7f68e3e0eaac0b2033d5/pytorch_lightning/core/lightning.py#L1533

File: pytorch_lightning/trainer/training_loop.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/trainer/training_loop.py:372]: I'm sure there is. This could get cleaned up. I'm just keeping the current logic in this PR

https://github.com/PyTorchLightning/pytorch-lightning/blob/0e45220263f4e2045dfe7f68e3e0eaac0b2033d5/pytorch_lightning/trainer/training_loop.py#L701-L702

I would wait until more legacy code is removed though

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]
